clang-tools-extra/docs/clang-tidy/checks/readability/avoid-return-with-void-value.rst

```diff
@@ -0,0 +1,30 @@
+.. title:: clang-tidy - readability-avoid-return-with-void-value
+
+readability-avoid-return-with-void-value
+========================================
+
+Complains about statements returning expressions of type ``void``. It can be
+confusing if a function returns an expression even though its return type is
+``void``.
+
```

Comment:
Add info that in "C" those issues are detected by -Wpedantic warning.
`warning: void function 'f' should not return void expression [-Wpedantic]`